src/object/cli_obj.c

```diff
@@ -1477,7 +1478,8 @@ shard_task_sched(tse_task_t *task, void *arg)
 		 * Also retry the shard IO if it got retryable error last time.
 		 */
 		target = obj_shard2tgtid(shard_auxi->obj, shard_auxi->shard);
-		if (obj_retry_error(task->dt_result) ||
+		if (obj_auxi->req_tgts.ort_srv_disp ||
```

Comment:
if the request need to be sent to server to dispatch, then for retry need to resend the request, it actually covered by "obj_retry_error(task->dt_result)", adding "obj_auxi->req_tgts.ort_srv_disp" just for clear.

the logic here was for cli-dispatch replica, some cases only need to retry partial shards' request (when shard's target changed) rather than retry all shards' request.
i thought just now that when epoch changed, even when shard's target not changed we also need to retry the shard request. I'll refresh it.

Reply:
Then how to handle the IO retry case with part of shards under client dispatch mode? There seems something to be improved, for example: if some shards return retry failure, but others return non-retry error, then it seems NOT necessary to retry, right? I mean under client dispatch mode.

Reply:
agreed, seems the detailed code was changed/broken, i'll check and change, thanks.

Reply:
indeed, we have to clean this up